cmd/junitreport/pkg/parser/interfaces.go

```diff
@@ -0,0 +1,61 @@
+package parser
+
+import (
+	"bufio"
+
+	"github.com/openshift/origin/cmd/junitreport/pkg/api"
+)
+
+// TestOutputParser knows how to parse test output to create a collection of test suites
+type TestOutputParser interface {
+	Parse(input *bufio.Scanner) (*api.TestSuites, error)
+}
+
+// TestDataParser knows how to take raw test data and either characterize it as data that
+// contains useful information or extract the useful information from it
+type TestDataParser interface {
```

Comment:
not sure the `ContainsX`, `ExtractX` pattern is useful... doesn't that mean you're doing work twice, and would I ever check if a line contains something I wouldn't immediately want to extract?